psfws/gen_params.py

```
@@ -0,0 +1,385 @@
+
+        if telemetry_file is not None:
+            self._paths['telemetry'] = pathlib.Path.joinpath(psfws_base, 
+                                                             telemetry_file)
+            use_telemetry = True
+        else:
+            use_telemetry = False
+             
+        for file_path in self._paths.values():
+            if not file_path.is_file():
+                print(f'code running from: {psfws_base}')
+                raise FileNotFoundError(f'file {file_path} not found!')
+
+        # set ground altitude, turbulence pdf, and j/wind correlation
+        self.h0 = h_tel 
+        self.j_pdf = {k: utils.lognorm(v['s'], v['scale']) for k, v in turbulence.items()}
+        self.rho_jv = rho_jv
+
+        self._rng = np.random.default_rng(seed)
+
+        # load and match forecast/telemetry data
+        self._load_data(use_telemetry)
+
+        # if using correlation between wind speed and ground turbulence,
+        # draw values in advance and perform correlation of marginals
+        if self.rho_jv is not 0:
+            # draw JGL values
+            j_gl = self.j_pdf['gl'].rvs(size=self.N, random_state=self._rng)
+            # correlate and store modified dataframe
+            self.data_gl = utils.correlate_marginals(self.data_gl,
+                                                     j_gl,
+                                                     self.rho_jv,
+                                                     self._rng)
+
+    def _load_data(self, use_telemetry=True):
+        """Load data from forecast, telemetry files, match, and store."""
+        forecast = pickle.load(open(self._paths['forecast_data'], 'rb'))
+        forecast = utils.process_forecast(forecast)
+
+        try: # first, find forecast dates within the date range desired
+            forecast_dates = forecast.index
+        except KeyError:
+            print("Requested dates are not within range of available data!")
```

I'm confused by this.  
First, that you just swallow the error and don't reraise the exception.  Is it really ok that the dates are invalid and you can just print out a message and move on?  
And second, this function doesn't use the variable `forecast_dates` anywhere.  Does this action do something behind the scenes inside `forecast` to change its internal state somehow?  If so, probably worth a comment saying that that is the point.  If not, I guess you can remove this whole try/except block.

---

Also, wouldn't it be AttributeError that is thrown if the `try` block failed?

---

Thanks for catching this; I can remove the whole block, it is a remanent of a removed functionality.